.github/workflows/markdown-link-check.yml

```diff
@@ -6,7 +6,7 @@ on:
     branches: ["main"]
 
 permissions:
-  contents: read
+  contents: write
```

Comment:
What is the reason of using `write` permission for this workflow?

Reply:
it will write comments in a PR to point directly at the spot where a mistake is.

Reply:
I like to avoid comments from workflows in PRs since it triggers an email notification and also is mixed with comments from developers regarding actual code. Instead, these things can be checked in workflow logs when needed. But it's my personal preference, so feel free to decide.

Reply:
@markwallace-microsoft any preference here?

Reply:
Looks like the new workflow action doesn't output all links, but only failed ones, which should be easy to find and fix:
![image](https://github.com/user-attachments/assets/6d0644a6-ad12-4eff-a683-16d74eb1664a)